### PR TITLE
Add window theme change handler

### PR DESCRIPTION
**Pull request by levinli303**
When an app goes to the background, UIKit changes UIWindow's trait collection to the other theme (change to light when in dark, change to dark when in light) so both snapshots are preserved and can respond to theme changes in background in app switcher.

For this case, UIWindow's `traitCollectionDidChange(:_)` is called where UIScreen's `traitCollectionDidChange(:_)` is not invoked, so the themeChangeHandler is not called. Adding a window theme change handler that wires up to UIWindow's `traitCollectionDidChange(:_)` for needs like that.

### CODE REVIEW

File: Sources/DarkModeCore/DMTraitCollection.m [hunk 1]
```diff
@@ -311,3 +316,22 @@ - (DMTraitCollection *)dmTraitCollection {
...
+@end
+
+@implementation UIWindow (DMTraitEnvironment)
+
+- (void)dmTraitCollectionDidChange:(DMTraitCollection *)previousTraitCollection {
+  [super dmTraitCollectionDidChange:previousTraitCollection];
+
+  if (@available(iOS 13, *)) {
+    if (previousTraitCollection.userInterfaceStyle != self.dmTraitCollection.userInterfaceStyle && _windowThemeChangeHandler)
+      _windowThemeChangeHandler(self);
```
Comment (zhuorantan) [inline Sources/DarkModeCore/DMTraitCollection.m:331]: If there are multiple windows (iPad multi-window), would the `themeChangeHandler()` be called multiple times?

Comment (levinli303) [reply Sources/DarkModeCore/DMTraitCollection.m:331]: yes, it is called per window, I think set a new value to `overrideUserInterfaceStyle` will also invoke it. so a UIWindow parameter is passed in the block to distinguish between windows.

Comment (zhuorantan) [reply Sources/DarkModeCore/DMTraitCollection.m:331]: I think we'd better to point this out explicitly in the method documentation, and explain what kind of functionality should go into this handler.

Comment (levinli303) [reply Sources/DarkModeCore/DMTraitCollection.m:331]: Agreed, I'll tell the difference in detail in the section for setting up FluentDarkModeKit in readme

Comment (levinli303) [reply Sources/DarkModeCore/DMTraitCollection.m:331]: updated.

**Review by zhuorantan — APPROVED** [review]

**Pull request merged**